ci/docker/common.yaml

```diff
@@ -46,3 +46,6 @@ packages:
     # Force git as non-buildable to allow deprecated versions in environments
     # https://github.com/spack/spack/pull/30040
     buildable: false
+  pika:
+    require:
+      - '@git.release-0.30.X=0.30.0'
```

Comment:
To do:
- bump minimum required pika version in spack package to 0.30.0 (or 0.31.0 depending on what issues we still find)
- bump spack commit to include at least 0.30.0
- remove this `require`